src/main/java/com/example/demo/service/PostsService.java

```diff
@@ -88,4 +95,83 @@ public Post unLike(Long id) {
         post.setLikeCount(max(0, post.getLikeCount() - 1));
         return postsRepository.save(post);
     }
+
+    // ------------------------------------------------------------------------------------------------------------------------------
+    // Function to check report Spam
+    public String spam(String description) {
+        Map<String, Integer> mp = new TreeMap<>();
```

Comment:
Duplicate logic tách từ ở cả 2 function `spam()` và `impolite()`.
Nên bỏ ra ngoài function

Reply:
Ngoài ra, em thử suy nghĩ xem, trong bài toán của mình,  `TreeMap` hay `HashMap` tối ưu hơn?